notes/2015-05-21.caching_and_versions.md

```diff
@@ -0,0 +1,54 @@
+# 2015-05-21
+I'm working on implementing a simple caching strategy and I'm trying to work out the best strategy for doing cache invalidation when deploying a new version of a model.  It seems like a model should know it's version.  I wonder if version should be the UTC timestamp at which a model was generated.  Redis can handle a simple set of types, so we could save space and potentially improve performance by storing it as an INT rather than a string.  The problem with version numbers is that they don't order the way we'd like.  E.g.:
+
+    >>> "1.9.0" > "1.25.0"
+    True
+
+OK.  Let me list the options:
+
+* UTC timestamp of model generation (in revscoring.ScorerModel)
+** Pros:
+*** No need to remember to update the version name when generating a new model
```

Comment:
This list looks broken at
https://github.com/halfak/Objective-Revision-Evaluation-Service/blob/caching/notes/2015-05-21.caching_and_versions.md